nbt/src/main/java/net/md_5/bungee/nbt/limit/NbtLimiter.java

```diff
@@ -0,0 +1,63 @@
+package net.md_5.bungee.nbt.limit;
+
+import lombok.RequiredArgsConstructor;
+import net.md_5.bungee.nbt.exception.NbtLimitException;
+
+@RequiredArgsConstructor
+public class NbtLimiter
+{
+    private static final int MAX_STACK_DEPTH = 512;
+
+    private final long maxBytes;
```

Comment:
given int max > 2gb, do we need a long here?

Reply:
yes we do for generall overflow preventions, vanilla does the same, most nbt libs don't and they can all be exploited and crashed

Reply:
but i also used the math.multiplyExact and addExact so in case it happens that a long will still overflow an exception will occour and nobody is able to allocate big arrays by letting numbers overflow

Reply:
i mean because these methods are used, it'd be enough to use int